src/main/java/baseball/Service/ScoreService.java

```diff
@@ -0,0 +1,49 @@
+package baseball.Service;
+
+import baseball.Constant.BallStatus;
+import baseball.domain.BaseBallNumbers;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.stream.IntStream;
+
+public class ScoreService {
+
+  public static final int START_NUMBERS_INCLUSIVE = 0;
+  public static final int BASEBALL_NUMBERS_SIZE = 3;
+  private static BaseBallNumbers answerNumbers;
+
+  public ScoreService(BaseBallNumbers answerNumbers) {
+    this.answerNumbers = answerNumbers;
+  }
+
+  public Boolean isStrike(BaseBallNumbers playerNumbers, int index) {
+    return playerNumbers.getNumbers(index) == answerNumbers.getNumbers(index);
+  }
+
+  public Boolean isBall(BaseBallNumbers playerNumbers, int index) {
+    return (playerNumbers.getNumbers(index) != answerNumbers.getNumbers(index)
+    ) && (answerNumbers.getNumbers().stream()
+        .anyMatch(x -> x.getBallNumber() == playerNumbers.getNumbers(index)));
+  }
+
+  // TODO : nothing도 따로 메소드로 관리하는게 나을까?
+  public Boolean isNothing(BaseBallNumbers playerNumbers, int index) {
+    return answerNumbers.getNumbers().stream()
+        .allMatch(x -> x.getBallNumber() != playerNumbers.getNumbers(index));
+  }
+
+  public HashMap<BallStatus, Integer> countScore(BaseBallNumbers playerNumbers) {
+    // TODO : 더 좋은 방법은 없을까? + 따로 메소드 만들어서 해도 됐을듯. + 매직넘버 사용 위가 나을까 아래가 나을까?
+    int countStrike = (int) IntStream.range(START_NUMBERS_INCLUSIVE, playerNumbers.getNumbers().size())
+
+        .filter(i -> isStrike(playerNumbers, i))
+        .count();
+    int countBall = (int) IntStream.range(START_NUMBERS_INCLUSIVE, BASEBALL_NUMBERS_SIZE)
+        .filter(i -> isBall(playerNumbers, i))
+        .count();
+    LinkedHashMap<BallStatus, Integer> countMap = new LinkedHashMap<>();
```

Comment:
변수 선언시 구현체보다 인터페이스로 선언하는 것도 좋다는 생각이 듭니다!
```suggestion
    Map<BallStatus, Integer> countMap = new LinkedHashMap<>();
```

Reply:
인터페이스 선언! 기억해놓겠습니다!
그걸 위해서는 컬렉션 프레임워크의 구조도 기억해놔야겠어요..
여담이지만 자바를 잘 사용하려면 상속구조를 잘 공부해놓는게 중요한 것 같네요